### DISCUSSION TITLE
DataGrid with Remote Data and Pagination Not Resetting After Filtering Data With Reduced Result Set

### DISCUSSION THREAD

**Discussion by NickHirras** [microsoft/fluentui-blazor#1575] [Q&A]
This is a blazor wasm application. I'm using a FluentDataGrid, fetching remote data with ItemsProvider, and utilizing Paginationpu
It's working great when filtering / paging through data. 

When the user changes search filters, resulting in fewer pages of results than the page they are currently viewing (for instance, they were previously viewing page 5 of results, and update a filter field reducing the results to 1 page).  

I can see behind the scenes, the pagination component is checking this condition, and triggering a second fetch, with Page set to a valid number in the new result size.  

https://github.com/microsoft/fluentui-blazor/blob/8ecf0f221aeb90b9e599885c8217cdf75f5384fb/src/Core/Components/Pagination/PaginationState.cs#L64-L69

I can see the retrieval happen in my browser, however the pagination UI isn't updating to show the correct state (it still shows old record count, and page / pages values).

**Comment by vnbaaij** [comment]
This is fixed in a recent PR and will be on the next release

**Reply by thieng101** [reply]
Hello, I want to ask if this fix has been released yet.

**Reply by NickHirras** [reply]
Yes, it was released in one of the March 2024 releases.
